runtime-modules/storage/src/data_directory.rs

```diff
@@ -316,58 +316,25 @@ decl_module! {
         /// Maximum objects allowed per inject_data_objects() transaction
         const MaxObjectsPerInjection: u32 = T::MaxObjectsPerInjection::get();
 
-        /// Adds the content to the system. Member id should match its origin. The created DataObject
-        /// awaits liaison to accept or reject it.
-        #[weight = 10_000_000] // TODO: adjust weight
-        pub fn add_content_as_member(
-            origin,
-            member_id: MemberId<T>,
-            content: Vec<ContentParameters<T::ContentId, DataObjectTypeId<T>>>
-        ) {
-            T::MemberOriginValidator::ensure_actor_origin(
-                origin,
-                member_id,
-            )?;
-
-            Self::ensure_content_is_valid(&content)?;
-
-            let owner = StorageObjectOwner::Member(member_id);
-
-            let owner_quota = Self::get_quota(&owner);
-
-            // Ensure owner quota constraints satisfied.
-            // Calculate upload voucher
-            let upload_voucher = Self::ensure_owner_quota_constraints_satisfied(owner_quota, &content)?;
-
-            // Ensure global quota constraints satisfied.
-            Self::ensure_global_quota_constraints_satisfied(upload_voucher)?;
-
-            let liaison = T::StorageProviderHelper::get_random_storage_provider()?;
-
-            //
-            // == MUTATION SAFE ==
-            //
-
-            // Let's create the entry then
-            Self::upload_content(owner_quota, upload_voucher, liaison, content.clone(), owner.clone());
-
-            Self::deposit_event(RawEvent::ContentAdded(content, owner));
-        }
-
-        /// Adds the content to the system. Requires root privileges. The created DataObject
+        /// Adds the content to the system. The created DataObject
         /// awaits liaison to accept or reject it.
         #[weight = 10_000_000] // TODO: adjust weight
         pub fn add_content(
             origin,
-            abstract_owner: AbstractStorageObjectOwner<ChannelId<T>, DAOId<T>>,
+            owner: StorageObjectOwner<MemberId<T>, ChannelId<T>, DAOId<T>>,
             content: Vec<ContentParameters<ContentId<T>, DataObjectTypeId<T>>>
         ) {
-            ensure_root(origin)?;
+            if let StorageObjectOwner::Member(member_id) = owner {
```

Comment:
I think this will be needed in many places, right? so we may as well factor it out into a `ensure_is_storage_object_owner(origin, owner)`

Reply:
Good point, it can be reused among storage uploading and further deletion extrinsics.  https://github.com/Joystream/joystream/pull/2134/commits/84ce5b9d251b730709d04f999f7d7444c1c2c18f